### PR TITLE
Prettier pagination

### LINKED ISSUE #169: Prettier Pagination
For the Events and Leaderboard pages

### PR DESCRIPTION
**Pull request by alexzhang1618**
# Info

Closes #169 

[description]

## Changes

- [Fill in here]

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) 

# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally on Desktop.
- [ ] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [ ] I have performed a self-review of my own code.
- [ ] I have followed the style guidelines of this project.
- [ ] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [ ] My changes produce no new warnings.

# Screenshots

### CODE REVIEW

File: src/components/common/PaginationControls/style.module.scss [hunk 1]
```diff
@@ -1,11 +1,14 @@
+@use 'src/styles/vars.scss' as vars;
+
 .paginationBtns {
   align-items: center;
   display: flex;
-  gap: 20px;
+  gap: 1.5rem;
   justify-content: center;
 
   .paginationBtn {
```
Comment (SheepTester) [inline src/components/common/PaginationControls/style.module.scss:7]: image

maybe you should make these a bit larger so they're the same size as the page numbers

File: src/components/common/PaginationControls/index.tsx [hunk 2]
```diff
@@ -9,12 +9,18 @@ interface PaginationControlsProps {
...
 
-const PaginationControls = ({ page, onPage, pages }: PaginationControlsProps) => {
-  const [value, setValue] = useState(String(page + 1));
+function calculatePagesToDisplay(page: number, totalPages: number, maxWidth: number): number[] {
+  const radius = (maxWidth - 1) / 2;
+  if (page + radius >= totalPages) {
+    return Array.from({ length: maxWidth }, (_, key) => totalPages - maxWidth + 1 + key);
+  }
+  const displayPage = page + 1;
+  const start = Math.max(1, displayPage - radius);
```
Comment (SheepTester) [inline src/components/common/PaginationControls/index.tsx]: i think you mean

```suggestion
  const start = Math.min(1, displayPage - radius);
```

**Review by SheepTester — CHANGES_REQUESTED** [review]
page numbers are broken on leaderboard:

image


image

Should we make the current number an input to still allow people to enter in a page number. like if they hover/focus on the blue number, the input reveals a border/background color

image


Maybe it could show nothing if there's only one page

File: src/components/common/PaginationControls/style.module.scss (hunk 1 above)
Comment (SheepTester) [reply src/components/common/PaginationControls/style.module.scss:7]: actually nvm i like that it;s smaller

**Comment by alexzhang1618** [comment]
Screenshot 2024-04-13 at 12 37 06 PM
Switching to a simpler, redesigned version of pagination

**Review by SheepTester — APPROVED** [review]
lgtm

**Pull request merged**